src/app/client/src/app/modules/learn/components/batch/update-course-batch/update-course-batch.component.ts

```diff
@@ -628,6 +661,70 @@ export class UpdateCourseBatchComponent implements OnInit, OnDestroy, AfterViewI
     }
   }
 
+  generateDataForDF() {
+    this.fetchForumIdReq = {
+      type: 'batch',
+      identifier: [this.batchId]
+    };
+  }
+
+  fetchForumConfig() {
+    this.fetchForumConfigReq = [{
+      type: 'batch',
+      identifier: this.batchId
+  }];
+    const subType = 'batch';
+    this.discussionService.fetchForumConfig(subType).subscribe((formData: any) => {
+      this.createForumRequest = formData[0];
+      this.createForumRequest['category']['context'] =  this.fetchForumConfigReq;
+    }, error => {
+      this.toasterService.error(this.resourceService.messages.emsg.m0005);
+    });
+  }
+
+  checkEnableDiscussions(batchId) {
+    if (this.batchUpdateForm.value.enableDiscussions === 'true') {
+      this.discussionService.createForum(this.createForumRequest).subscribe(resp => {
+        this.handleEnableDFChange('Yes');
+        this.toasterService.success('Enabled discussion forum successfully');
```

Comment:
here also the label should be used like 
this.toasterService.error(this.resourceService.messages.emsg.m0005);